animation_nodes/nodes/sound/MIDI_File_Parser.py

```diff
@@ -0,0 +1,55 @@
+import os
+import bpy
+from ... base_types import AnimationNode
+from ... utils.midi import MIDI_ParseFile
+
+# path : last modification, content
+cache = {}
+
+class MIDIFileParserNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_MidiFileParserNode"
+    bl_label = "MIDI File Parser"
+    bl_width_default = 180
+    errorHandlingType = "EXCEPTION"
+
+    def create(self):
+        self.newInput("Text", "Path", "path", showFileChooser = True)
+        self.newOutput("MIDITrackList", "Tracks", "tracks")
+
+    def draw(self, layout):
+        if self.inputs[0].isUnlinked:
+            name = os.path.basename(self.inputs[0].value)
+            if name != "":
+                layout.label(text = name, icon = "FILE_TEXT")
+
+    def drawAdvanced(self, layout):
+        self.invokeFunction(layout, "clearCache", text = "Clear Cache")
+
+    def clearCache(self):
+        cache.clear()
+
+    def execute(self, path):
+        if not os.path.exists(path):
+            self.raiseErrorMessage("Path does not exist")
+
+        key = path
+        lastModification = os.stat(path).st_mtime
+
+        loadFile = False
+        if key not in cache:
+            loadFile = True
+        else:
+            oldLastModification = cache[key][0]
+            if lastModification > oldLastModification:
+                loadFile = True
+
+        if loadFile:
+            try:
+                tracks = MIDI_ParseFile(path)
+                cache[key] = (lastModification, tracks)
+            except LookupError:
+                self.raiseErrorMessage("Invalid Encoding")
+        else:
+            tracks = None
```

Comment:
Is this necessary?

Reply:
I take text_file_reader.py as an example. I think we have to imagine user made a bad choice of file

Reply:
I guess we can use an LRU cache like in the sound nodes to simplify things.

And why do we have the line `tracks = None`?

Reply:
I can't find a model in sound nodes. Nor in sound_from_sequence or sound_spectrum I found something about LRU cache.

Reply:
See the `getCachedSoundDataFromPath` function in `sound_sequence.py`.

Reply:
This is the last code in this correction series where I work. It's more difficult for me.

Reply:
Ok, for now I push all my commits but the lrucache part is still missing which I have no control at all. I need help.

Reply:
Alright. I will finish the things at hand and get back to you.